app/controllers/api/v2/access_requests_controller.rb

```diff
@@ -0,0 +1,34 @@
+module API
+  module V2
+    class AccessRequestsController < API::V2::ApplicationController
+      before_action :build_access_request
+      before_action :build_requesting_user
+
+      def approve
+        if @access_request.admin?(@current_user) && requesting_user_belongs_to_oranisation?
+          @access_request.completed!
+          @access_request.update_access(@access_request, @requesting_user)
+        end
+      end
+
+      def decline
+        @access_request.declined! if @access_request.admin?(@current_user)
+      end
+
+    private
+
+      def build_access_request
+        @access_request = AccessRequest.find_by(id: params[:id])
```

Comment:
Not a big deal or anything but this can be one line
`@access_request = authorise AccessRequest.find_by(id: params[:id])`